src/datamodel.js

```diff
@@ -695,6 +725,128 @@ class DataModel extends Relation {
+     * @param {Object} config - The configuration object
+     * @param {string} [config.saveChild] - Configuration to save child or not
+     * @param {string}[config.mode=FilteringMode.NORMAL] -The mode of the selection.
+     * @return {Array}  Returns the new DataModel instances after operation.
+     */
+    splitByRow (dimensionArr, reducerFn, config) {
+        const fieldsConfig = this.getFieldsConfig();
+
+        dimensionArr.forEach((fieldName) => {
+            if (!fieldsConfig[fieldName]) {
+                throw new Error(`Field ${fieldName} doesn't exist in the schema`);
+            }
+        });
+
+        const defConfig = {
+            mode: FilteringMode.NORMAL,
+            saveChild: true
+        };
+
+        config = Object.assign({}, defConfig, config);
+
+        return splitWithSelect(this, dimensionArr, reducerFn, config);
+    }
+
+    /**
+     * Creates a set of new {@link DataModel} instances by splitting the set of fields in the source {@link DataModel}
+     * instance based on a set of common and unique field names provided.
+     *
+     * Each DataModel created contains a set of fields which are common to all and a set of unique fields.
+     * It also accepts configurations such as saveChild and mode(inverse or normal) to include/exclude the respective
+     * fields
+     *
+     * @example
+     *  // without predicate function:
+     *  const splitDt = dt.splitByColumn( [['Acceleration'], ['Horsepower']], ['Origin'])
+     *  console.log(splitDt));
+     * // This should give two unique DataModel instances, both having the field 'Origin' and
+     * // one each having 'Acceleration' and 'Horsepower' fields respectively
+     *
+     * @example
+     *  // without predicate function:
+     *  const splitDtInv = dt.splitByColumn( [['Acceleration'], ['Horsepower'],['Origin', 'Cylinders'],
+     *                           {mode: 'inverse'})
+     *  console.log(splitDtInv));
+     * // This should give DataModel instances in the following way:
+     * // All DataModel Instances do not have the fields 'Origin' and 'Cylinders'
+     * // One DataModel Instance has rest of the fields except 'Acceleration' and the other DataModel instance
+     * // has rest of the fields except 'Horsepower'
+     *
+     *
+     *
+     * @public
+     *
+     * @param {Array} uniqueFields - Set of unique fields included in each datamModel instance
+     * @param {Array} commonFields - Set of common fields included in all datamModel instances
+     * @param {Object} config - The configuration object
+     * @param {string} [config.saveChild] - Configuration to save child or not
+     * @param {string}[config.mode=FilteringMode.NORMAL] -The mode of the selection.
+     * @return {Array}  Returns the new DataModel instances after operation.
+     */
+    splitByColumn (uniqueFields = [], commonFields = [], config) {
+        const defConfig = {
```

Comment:
define default configs in defaults file

Reply:
Have written in the format that is written in the dataModel code. check the code for `select` in `relation.js`